lib/buffer.js

```diff
@@ -289,6 +289,24 @@ Buffer.prototype.compare = function compare(b) {
 };
 
 
+Buffer.prototype.indexOf = function indexOf(val, byteOffset) {
+  if (byteOffset > 0xffffffff)
+    byteOffset = 0xffffffff;
+  else if (byteOffset < 0)
+    byteOffset = 0;
+  byteOffset = byteOffset >>> 0;
+
+  if (typeof val === 'string')
+    return internal.indexOfString(this, val, byteOffset);
+  if (val instanceof Buffer)
+    return internal.indexOfBuffer(this, val, byteOffset);
```

Comment:
Ok. I still find it funny to find an empty string within an empty string at index 0 :)